validator/validate_api.py

```diff
@@ -72,21 +66,28 @@ def setup_parse_and_data(setup_state):
 
 def get_question_data_by_key(key, val):
```

Comment:
Maybe add some kind of comment warning that you should only query by id-like attributes and this method will return nonsense if your query returns more than 1 question?

Reply:
Not non-sense - it _will_ be a page that's associated with that question id, in some way, so will help. Just not as much as it should. You see now why I would like tutor to pass the book (&page) context …